### PR TITLE
Branches code to work with .lg files from CL-UI to CL-Models

### PR DESCRIPTION
**Pull request by tpmsr**
Pulls LG file parsing code from ConversationLearner-UI.

Tested via local integration, also adds unit tests to the parsing function.

This is pre-work for creation of a new .transcript conversion tool.

### CODE REVIEW

File: src/OBIUtils.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+const findLastOrThrow = (haystack: string, needle: string): number => {
+    return findOrThrow(haystack.lastIndexOf.bind(haystack), needle)
+}
+
+/**
+ * Parses OBI .dialog files from custom format into structured data.
+ */
+export class ObiUtils {
+
+    public static parseLGString(text: string): Map<string, LGItem> {
```
Comment (LarsLiden) [inline src/OBIUtils.ts]: One difference with this version is that it creates a new Map.  The previous version appends to an existing map.  In the sample you have there's only on LG file, but there could me many

File: src/OBIUtils.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        let output = new Map<string, LGItem>()
```
Comment (LarsLiden) [inline src/OBIUtils.ts]: would prefer a more descriptive name than "output" (something like lgMap)

**Comment by tpmsr** [comment]
Ah, Windows.  Looks like some file case issues are causing tests to fail.  wip.

File: src/OBIUtils.ts (hunk 1 above)
Comment (tpmsr) [reply src/OBIUtils.ts]: Is that a problem ?  It's reasonably straightforward to create the union of several Maps

`const union = new Map([...a, ...b, ...c])`

Happy to change it to take the Map parameter if that seems more straightforward.

File: src/OBIUtils.ts (hunk 2 above)
Comment (tpmsr) [reply src/OBIUtils.ts]: Done

File: src/OBIUtils.ts (hunk 1 above)
Comment (LarsLiden) [reply src/OBIUtils.ts]: Computationally it more intense, but fine.  :)

Comment (tpmsr) [reply src/OBIUtils.ts]: Actually, my proposal doesn't work with an array of Maps.  Still learning where the rough edges of Typescript are.

Updated to take input Map param.

File: src/ObiUtils.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            const backticksStart = findFirstOrThrow(item, "```")
+            const suggestionsStart = findLastOrThrow(item, "[")
+            if (backticksStart > suggestionsStart) throw new RangeError('``` not found')
+            const body = item.substring(backticksStart + 3, suggestionsStart).trim()
+            const suggestionTag = "[Suggestions="
+            const suggestionList = item.substring(
+                findLastOrThrow(item, suggestionTag) + suggestionTag.length,
+                findLastOrThrow(item, "]"))
+            const suggestions = suggestionList.length > 0 ? suggestionList.split('|') : []
+            const element = {
```
Comment (LarsLiden) [inline src/ObiUtils.ts]: I'd call this lgItem instead of item, and do a type declaration:

const lgItem: LGItem =

Comment (tpmsr) [reply src/ObiUtils.ts]: Done!

**Pull request merged**